rest/route/host_spawn_test.go

```diff
@@ -28,221 +28,268 @@ import (
 )
 
 func TestHostPostHandler(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
+	checkUnexpirableHostSchedule := func(t *testing.T, h *host.Host, opts host.SleepScheduleOptions) {
+		assert.True(t, h.NoExpiration)
+		assert.Equal(t, opts.WholeWeekdaysOff, h.SleepSchedule.WholeWeekdaysOff)
+		assert.Equal(t, opts.DailyStartTime, h.SleepSchedule.DailyStartTime)
+		assert.Equal(t, opts.DailyStopTime, h.SleepSchedule.DailyStopTime)
+		assert.Equal(t, opts.TimeZone, h.SleepSchedule.TimeZone)
+		assert.NotZero(t, h.SleepSchedule.NextStartTime)
+		assert.NotZero(t, h.SleepSchedule.NextStopTime)
+	}
 
-	require.NoError(t, db.ClearCollections(distro.Collection, host.Collection))
-
-	env := &mock.Environment{}
-	assert.NoError(t, env.Configure(ctx))
-	env.EvergreenSettings.Spawnhost.SpawnHostsPerUser = 10
-	env.EvergreenSettings.Spawnhost.UnexpirableHostsPerUser = 5
-	var err error
-	env.RemoteGroup, err = queue.NewLocalQueueGroup(ctx, queue.LocalQueueGroupOptions{
-		DefaultQueue: queue.LocalQueueOptions{Constructor: func(context.Context) (amboy.Queue, error) {
-			return queue.NewLocalLimitedSize(2, 1048), nil
-		}}})
-	assert.NoError(t, err)
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T, env *mock.Environment, rh *hostPostHandler, u *user.DBUser, d *distro.Distro){
+		"SpawnsHostForTask": func(ctx context.Context, t *testing.T, env *mock.Environment, rh *hostPostHandler, u *user.DBUser, d *distro.Distro) {
```

Comment:
Small nit to just pass testing.T and then use t.Context() instead of passing another context

Reply:
We need to pass in an explicit context due to `gimlet.AttachUser`, which makes a child context.